nautobot/core/apps/__init__.py

```diff
@@ -0,0 +1,247 @@
+from django.apps import AppConfig
+from django.urls import reverse
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+from nautobot.utilities.choices import ButtonActionColorChoices
+
+
+registry["nav_menu"] = {"tabs": {}}
+
+
+class NautobotConfig(AppConfig):
+    """
+    Custom AppConfig for Nautobot application.
+
+    Adds functionality to generate the HTML navigation menu using `navigation.py` files from nautbot
+    applications.
+    """
+
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        """
+        Ready function initiates the import application.
+
+        As the application can be loaded from either the root of the app or `apps.py`, the __module__
+        name needs to be modified to all the ready function to import the menu items.
+
+        To import the menu items the `.apps` needs to be removed from the __module__ name if it
+        exists.
+
+        Examples:
+            Loaded from application root:   `nautobot.circuits`
+            Loaded from `app.py`:           `nautobot.circuits.apps`
+        """
+        if self.__module__.endswith(".apps"):
+            module_location = self.__module__[:-5]
+        else:
+            module_location = self.__module__
+        menu_items = import_object(f"{module_location}.{self.menu_tabs}")
```

Comment:
Actually, I think this could perhaps be done more gracefully by using `self.name` (https://docs.djangoproject.com/en/3.2/ref/applications/#django.apps.AppConfig.name) - this *should* work:
```suggestion
        menu_items = import_object(f"{self.name}.{self.menu_tabs}")
```